### PR TITLE
Add setMetaData and getMetaData

### PR DESCRIPTION
**Pull request by mihaieremia**
Added a view function to get the metadata.
Added a modal to show the metadata or set/update the fields via ```setMetaData``` function.
https://testnet-explorer.elrond.com/transactions/16d36a1de064a7770438e49571a372c5cc12aa1cc3905f93aec6bb314d3ddc96
Validation for URL input.
image

image
image

### CODE REVIEW

File: react-delegationdashboard/src/components/Overview/Header/SetAgencyMetaDataModal.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            </p>
+            <Formik
+              initialValues={agencyMetaData}
+              onSubmit={values => {
+                handleReDelegationCapActivation(values);
+              }}
+              validationSchema={object().shape({
+                website: string()
+                  .required('Required')
+                  .test('URL', 'URL is not valid!', value => {
```
Comment (danielailie) [inline react-delegationdashboard/src/components/Overview/Header/SetAgencyMetaDataModal.tsx:54]: can you remove the validation for https, it is ok to let the user only add the site without the https://

**Review by danielailie — CHANGES_REQUESTED** [review]

File: react-delegationdashboard/src/components/Layout/index.tsx [hunk 2]
```diff
@@ -2,8 +2,9 @@ import { QueryResponse } from '@elrondnetwork/erdjs/out/smartcontracts/query';
 import denominate from 'components/Denominate/formatters';
 import { denomination, decimals } from 'config';
 import { useContext, useDispatch } from 'context';
+import { emptyAgencyMetaData } from 'context/state';
 import { contractViews } from 'contracts/ContractViews';
-import { ContractOverview, NetworkConfig, NetworkStake, Stats } from 'helpers/types';
+import { ContractOverview, AgencyMetadata, NetworkConfig, NetworkStake, Stats } from 'helpers/types';
```
Comment (arhtudormorar) [inline react-delegationdashboard/src/components/Layout/index.tsx:7]: maybe use typescript IBeforeInterface notation to identify types in project. Or NetworkConfigType (suffix with Type at the end)

File: react-delegationdashboard/src/components/Layout/index.tsx [hunk 3]
```diff
@@ -41,8 +42,19 @@ const Layout = ({ children, page }: { children: React.ReactNode; page: string })
     );
   };
 
+  const getAgencyMetaDataType = (value: QueryResponse) => {
+    if(value.returnData.length === 0) {
```
Comment (arhtudormorar) [inline react-delegationdashboard/src/components/Layout/index.tsx:46]: be sure value is defined. if value === undefined => Exception

File: react-delegationdashboard/src/components/Overview/Header/SetAgencyMetaDataModal.tsx (hunk 1 above)
Comment (arhtudormorar) [reply react-delegationdashboard/src/components/Overview/Header/SetAgencyMetaDataModal.tsx:54]: I would suggest using https://developer.mozilla.org/en-US/docs/Web/API/URL/URL. If `new URL(value)` fails then false, else true

File: react-delegationdashboard/src/components/Overview/Header/index.tsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import { Link, useLocation } from 'react-router-dom';
 import { Address } from '@elrondnetwork/erdjs/out';
 import { useContext } from '../../../context';
```
Comment (arhtudormorar) [inline react-delegationdashboard/src/components/Overview/Header/index.tsx:4]: better use absolute imports

File: react-delegationdashboard/src/components/Layout/index.tsx (hunk 2 above)
Comment (mihaieremia) [reply react-delegationdashboard/src/components/Layout/index.tsx:7]: Isn't the same for ```ContractOverview``` ?

File: react-delegationdashboard/src/components/Overview/Header/SetAgencyMetaDataModal.tsx (hunk 1 above)
Comment (mihaieremia) [reply react-delegationdashboard/src/components/Overview/Header/SetAgencyMetaDataModal.tsx:54]: ```new URL()``` is still asking to use ```https://``` or ```http://```

Comment (mihaieremia) [reply react-delegationdashboard/src/components/Overview/Header/SetAgencyMetaDataModal.tsx:54]: I added a new regex that handles all cases ```www.test.com/test.com/http(s)://test.com/http(s)://www.test.com```

**Review by danielailie — COMMENTED** [review]
:+1:

**Review by danielailie — APPROVED** [review]
:+1:

File: react-delegationdashboard/src/components/Layout/index.tsx (hunk 2 above)
Comment (arhtudormorar) [reply react-delegationdashboard/src/components/Layout/index.tsx:7]: OK

**Review by arhtudormorar — APPROVED** [review]

**Pull request merged**